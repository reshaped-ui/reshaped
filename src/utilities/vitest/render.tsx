import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
