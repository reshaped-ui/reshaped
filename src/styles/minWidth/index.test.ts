import { expect, test, describe } from "vitest";
import getMinWidthStyles from "./index";

describe("Styles/MinWidth", () => {
	test("handles px value", () => {
		expect(getMinWidthStyles("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(getMinWidthStyles("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(getMinWidthStyles(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getMinWidthStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getMinWidthStyles({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
