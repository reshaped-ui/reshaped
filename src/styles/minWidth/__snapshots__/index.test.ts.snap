// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles/MinWidth handles % value 1`] = `
{
  "classNames": [
    "--type-literal",
  ],
  "variables": {
    "--rs-min-w-s": "50%",
  },
}
`;

exports[`Styles/MinWidth handles px value 1`] = `
{
  "classNames": [
    "--type-literal",
  ],
  "variables": {
    "--rs-min-w-s": "50px",
  },
}
`;

exports[`Styles/MinWidth handles responsive value 1`] = `
{
  "classNames": [
    "--type-literal",
    "--type-literal--l",
  ],
  "variables": {
    "--rs-min-w-l": "50%",
    "--rs-min-w-s": "50px",
  },
}
`;

exports[`Styles/MinWidth handles undefined value 1`] = `null`;

exports[`Styles/MinWidth handles unit value 1`] = `
{
  "classNames": [
    "--type-unit",
  ],
  "variables": {
    "--rs-min-w-s": 5,
  },
}
`;
